contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,365 @@
-  ///
-  /// @param amount  The amount of nxm to unstake
-  /// @param to      The member address where the NXM is transfered to. Useful for switching
-  ///                membership during stake lockup period and thus allowing the user to withdraw
-  ///                their staked amount to the new address when possible.
-  function unstake(uint96 amount, address to) external override whenNotPaused {
-
-    uint stakeAmount = stakeOf[msg.sender].amount;
-    require(amount <= stakeAmount, InvalidAmount(stakeAmount));
+  function setGroupMetadata(uint groupId, bytes32 ipfsMetadata) override external onlyContracts(C_GOVERNOR) {
+    require(groupId > 0 && groupId <= _groupCount, InvalidGroupId());
 
-    uint govLockupExpiry = nxm.isLockedForMV(msg.sender);
-    require(block.timestamp >= govLockupExpiry || to == msg.sender, StakeLockedForGovernance(govLockupExpiry));
-
-    _unstake(msg.sender, amount, to);
+    _groupsMetadata[groupId] = ipfsMetadata;
+    emit GroupMetadataSet(groupId, ipfsMetadata);
   }
 
-  /// Withdraws all of the the given staker's stake
-  ///
-  /// @dev At least stakeLockupPeriodInDays must have passed since the last vote.
-  ///
-  /// @param staker  The address of the staker whose stake will be unstaked
-  function unstakeAllFor(address staker) external override whenNotPaused onlyTokenController {
-    _unstake(staker, stakeOf[staker].amount, staker);
+  function removeAssessorFromGroup(uint assessorMemberId, uint groupId) override external onlyContracts(C_GOVERNOR) {
+    require(groupId > 0 && groupId <= _groupCount, InvalidGroupId());
+
+    require(assessorMemberId != 0, InvalidMemberId());
+    _groups[groupId].remove(assessorMemberId);
+    _groupsForAssessor[assessorMemberId].remove(groupId);
+    emit AssessorRemovedFromGroup(groupId, assessorMemberId);
   }
 
-  function _unstake(address staker, uint96 amount, address to) internal {
+  function removeAssessorFromAllGroups(uint assessorMemberId) override external onlyContracts(C_GOVERNOR) {
+    require(assessorMemberId != 0, InvalidMemberId());
 
-    uint voteCount = votesOf[staker].length;
-    if (voteCount > 0) {
-      Vote memory latestVote = votesOf[staker][voteCount - 1];
-      uint assessmentLockupExpiry = latestVote.timestamp + STAKE_LOCKUP_PERIOD;
-      require(block.timestamp > assessmentLockupExpiry, StakeLockedForAssessment(assessmentLockupExpiry));
+    uint[] memory assessorsGroups = _groupsForAssessor[assessorMemberId].values();
+    uint assessorsGroupsLength = assessorsGroups.length;
+    for (uint groupIndex = 0; groupIndex < assessorsGroupsLength; groupIndex++) {
+      uint groupId = assessorsGroups[groupIndex];
+      _groups[groupId].remove(assessorMemberId);
+      emit AssessorRemovedFromGroup(groupId, assessorMemberId);
     }
 
-    stakeOf[staker].amount -= amount;
-    nxm.transfer(to, amount);
-
-    emit StakeWithdrawn(staker, to, amount);
+    _clearSet(_groupsForAssessor[assessorMemberId]._inner);
```

Comment:
This should really be part of the library itself. Given we're generally considering to copy over the libs instead of having external deps - we could copy over this one and add this function to it. Would also be very good to have tests for this lib since we're dealing with internals here.